### PR TITLE
Refactor predict output

### PR DESCRIPTION
**Pull request by suttang**
## Motivation and Context

This PR related to #480
To refactor the code around predict result output, we need to create `OutputWriter` and `OutputParser` I think.

In this PR, I created OutputWriter class.

## Description

- deleted save functions from predict.py
- created output writer class and test

## How has this been tested?

1. Compare output files from master and this PR's one.
2. `make test`

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature / Optimization (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: lmnet/lmnet/utils/predict_output/writer.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    def __init__(self, task, classes, image_size, data_format):
+        self.json_output = JsonOutput(task, classes, image_size, data_format)
+        self.image_from_json = ImageFromJson(task, classes, image_size)
+
+    def write(self, dest, outputs, raw_images, image_files, step, save_material=True):
+        """Save predict output to disk.
+           numpy array, JSON, and images if you want.
+
+        Args:
+            dest (str): paht to save file
```
Comment (iizukak) [inline lmnet/lmnet/utils/predict_output/writer.py]: typo in `paht` .

File: lmnet/lmnet/utils/predict_output/writer.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+        json = self.json_output(outputs, raw_images, image_files)
+        save_json(dest, json, step)
+
+        if save_material:
+            materials = self.image_from_json(json, raw_images, image_files)
+            save_materials(dest, materials, step)
+
+
+def save_npy(dest, outputs, step):
```
Comment (iizukak) [inline lmnet/lmnet/utils/predict_output/writer.py]: Why these function are not method of `OutputWriter` ?

File: lmnet/lmnet/utils/predict_output/writer.py [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    filepath = os.path.join(dest, "json", "{}.json".format(step))
+    os.makedirs(os.path.dirname(filepath), exist_ok=True)
+
+    with open(filepath, "w") as f:
+        f.write(json)
+
+    print("save json: {}".format(filepath))
+
+
+def save_materials(dest, materials, step):
```
Comment (iizukak) [inline lmnet/lmnet/utils/predict_output/writer.py]: `save_materials` looks general name. `save_images` is not enough?

File: lmnet/lmnet/utils/predict_output/writer.py (hunk 1 above)
Comment (suttang) [reply lmnet/lmnet/utils/predict_output/writer.py]: oops, thank you!

File: lmnet/lmnet/utils/predict_output/writer.py (hunk 3 above)
Comment (suttang) [reply lmnet/lmnet/utils/predict_output/writer.py]: One day we will come to deal with tasks other than images.
Named for that time.
If not, I will fix.

File: lmnet/lmnet/utils/predict_output/writer.py (hunk 2 above)
Comment (suttang) [reply lmnet/lmnet/utils/predict_output/writer.py]: I prefer "Don't put in a method if you don't need it" mentality.
These functions don't need the state of `OutputWriter`.
IMO, This makes these functions independent and easier to test.

File: lmnet/lmnet/utils/predict_output/writer.py (hunk 1 above)
Comment (suttang) [reply lmnet/lmnet/utils/predict_output/writer.py]: fixed!

File: lmnet/lmnet/utils/predict_output/writer.py (hunk 2 above)
Comment (iizukak) [reply lmnet/lmnet/utils/predict_output/writer.py]: I see.

File: lmnet/lmnet/utils/predict_output/writer.py (hunk 3 above)
Comment (iizukak) [reply lmnet/lmnet/utils/predict_output/writer.py]: I see. To the future.

Comment (suttang) [reply lmnet/lmnet/utils/predict_output/writer.py]: Go to the future!

File: lmnet/lmnet/utils/predict_output/writer.py [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+    if type(step) is not int:
+        raise ValueError("step must be integer.")
+
+    filepath = os.path.join(dest, "json", "{}.json".format(step))
+    os.makedirs(os.path.dirname(filepath), exist_ok=True)
+
+    with open(filepath, "w") as f:
+        f.write(json)
+
+    print("save json: {}".format(filepath))
```
Comment (tsawada) [inline lmnet/lmnet/utils/predict_output/writer.py]: we should not `print` debugging information.

File: lmnet/tests/lmnet_tests/util_tests/predict_output/conftest.py [hunk 5]
```diff
@@ -0,0 +1,10 @@
+import tempfile
+
+import pytest
+
+
+@pytest.fixture
+def temp_dir():
```
Comment (tsawada) [inline lmnet/tests/lmnet_tests/util_tests/predict_output/conftest.py]:
```
with TemporaryDirectory() as t:
    yield t.name
```

**Pull request merged**